src/ucp/core/ucp_am.c

```diff
@@ -951,6 +957,13 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nbx,
 
     UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
+    if (id > UINT16_MAX) {
```

Comment:
maybe 
```suggestion
    if (ENABLE_PARAMS_CHECK && ucs_unlikely(id > UINT16_MAX)) {
```
to avoid extra overhead on fast path, like it is done in other funcs in this file (e.g. `ucp_am_recv_data_nbx`)

Reply:
@brminich When is `ENABLE_PARAMS_CHECK` set to 0?

Reply:
it is a configure option which is set to `0` for release builds
```
# ./configure --help | grep params
  --disable-params-check  Disable checking user parameters passed to API,
# cat ./contrib/configure-release | grep params
        --disable-params-check \
```

Reply:
But I think we want to check this parameter in release as well. Don't we?

Reply:
I see. Thanks